src/function/builtin.rs

```diff
@@ -1,8 +1,25 @@
+use regex::Regex;
+
+use crate::error::*;
 use value::{FloatType, IntType};
 use EvalexprError;
 use Function;
 use Value;
 
+fn regex_with_local_errors(re_str: &str) -> Result<Regex, EvalexprError> {
```

Comment:
Do you think it would make sense to give `EvalexprError::InvalidRegex` a member that is a `regex::Error`? Then the user could handle his regex errors himself, and we don't impose our error messages on them. We can still use the formatting you propose in the `impl Display for EvalexprError`.

And then it would also make sense to rename `EvalexprError::InvalidRegex` to `EvalexprError::RegexError`, for generality.

Does `regex::Error` have more than these two variants?

Reply:
I agree this is a bit of complexity that it would have been good to hide somewhere. Unfortunately, `EvalexprError` derives `PartialEq`, but `Regex` does not implement it, so the compiler is not happy with a `RegexError(regex::Error)` variant.  Also, in order to make the regex crate optional, we would have to exclude that bit from the EvalexprError enum, which feels iffy. At this point those are the ones https://github.com/rust-lang/regex/blob/master/src/error.rs#L33 . Having studied the regex errors a bit more, it seems it is unnecessary to repeat the error message presentation, so I have simply removed the function and directly presents the `regex::Error` display message.

Reply:
That makes sense. Thanks for doing this research.